### PR TITLE
fix uniform memory issue

### PR DESCRIPTION
**Pull request by Mee-gu**

### CODE REVIEW

File: cocos/renderer/gfx-gles2/GLES2GPUObjects.h [hunk 1]
```diff
@@ -86,22 +86,60 @@ struct GLES2GPUInput {
...
+    name(rhs.name),
+    type(rhs.type),
+    stride(rhs.stride),
+    count(rhs.count),
+    size(rhs.size),
+    offset(rhs.offset),
+    gl_type(rhs.gl_type),
+    gl_loc(rhs.gl_loc)
+    {
+        CC_SAFE_FREE(buff);
```
Comment (minggo) [inline cocos/renderer/gfx-gles2/GLES2GPUObjects.h]: don't need to free in constructor

File: cocos/renderer/gfx-gles2/GLES2GPUObjects.h [hunk 2]
```diff
@@ -86,22 +86,60 @@ struct GLES2GPUInput {
...
+            name = rhs.name;
+            type = rhs.type;
+            stride = rhs.stride;
+            count = rhs.count;
+            size = rhs.size;
+            offset = rhs.offset;
+            gl_type = rhs.gl_type;
+            gl_loc = rhs.gl_loc;
+            CC_SAFE_FREE(buff);
+            buff = (uint8_t*)CC_MALLOC(size);
```
Comment (minggo) [inline cocos/renderer/gfx-gles2/GLES2GPUObjects.h]: don't need to realloc if size is the same

File: cocos/renderer/gfx-gles2/GLES2GPUObjects.h [hunk 3]
```diff
@@ -86,22 +86,60 @@ struct GLES2GPUInput {
...
+    stride(rhs.stride),
+    count(rhs.count),
+    size(rhs.size),
+    offset(rhs.offset),
+    gl_type(rhs.gl_type),
+    gl_loc(rhs.gl_loc)
+    {
+        CC_SAFE_FREE(buff);
+        buff = (uint8_t*)CC_MALLOC(size);
+        memcpy(buff, rhs.buff, size);
```
Comment (minggo) [inline cocos/renderer/gfx-gles2/GLES2GPUObjects.h]: can use `*this = rhs` to simplify the codes?

File: cocos/renderer/gfx-gles2/GLES2GPUObjects.h [hunk 4]
```diff
@@ -86,22 +86,60 @@ struct GLES2GPUInput {
...
+            binding = rhs.binding;
+            name = rhs.name;
+            type = rhs.type;
+            stride = rhs.stride;
+            count = rhs.count;
+            size = rhs.size;
+            offset = rhs.offset;
+            gl_type = rhs.gl_type;
+            gl_loc = rhs.gl_loc;
+            CC_SAFE_FREE(buff);
```
Comment (minggo) [inline cocos/renderer/gfx-gles2/GLES2GPUObjects.h]: use CC_FREE and CC_NEW instead to keep the same memory mechanism.

File: cocos/renderer/gfx-gles2/GLES2GPUObjects.h [hunk 5]
```diff
@@ -86,22 +86,60 @@ struct GLES2GPUInput {
...
+    uint binding;
+    String name;
+    GFXType type;
+    uint stride;
+    uint count;
+    uint size;
+    uint offset;
+    GLenum gl_type;
+    GLint gl_loc;
+    uint8_t* buff = nullptr;
```
Comment (minggo) [inline cocos/renderer/gfx-gles2/GLES2GPUObjects.h]: please set default value

File: cocos/renderer/gfx-gles2/GLES2GPUObjects.h [hunk 6]
```diff
@@ -86,22 +86,60 @@ struct GLES2GPUInput {
 };
 typedef vector<GLES2GPUInput>::type GLES2GPUInputList;
 
-struct GLES2GPUUniform {
```
Comment (minggo) [inline cocos/renderer/gfx-gles2/GLES2GPUObjects.h]: Should do the modification in GLES3GPUObjects.h too.

File: cocos/renderer/gfx-gles2/GLES2GPUObjects.h (hunk 4 above)
Comment (Mee-gu) [reply cocos/renderer/gfx-gles2/GLES2GPUObjects.h]: CC_NEW can not allocate specific bytes, it was used to new an Object.

File: cocos/renderer/gfx-gles2/GLES2GPUObjects.h [hunk 7]
```diff
@@ -86,22 +86,52 @@ struct GLES2GPUInput {
...
+            stride = rhs.stride;
+            count = rhs.count;
+            offset = rhs.offset;
+            gl_type = rhs.gl_type;
+            gl_loc = rhs.gl_loc;
+            if(size != rhs.size)
+            {
+                size = rhs.size;
+                CC_SAFE_FREE(buff);
+                buff = (uint8_t*)CC_MALLOC(size);
```
Comment (minggo) [inline cocos/renderer/gfx-gles2/GLES2GPUObjects.h]: should check the result

**Pull request merged**